libgnucash/engine/Account.cpp

```diff
@@ -3268,35 +3262,22 @@ gnc_account_get_full_name(const Account *account)
     /* errors */
     g_return_val_if_fail(GNC_IS_ACCOUNT(account), g_strdup(""));
 
-    /* optimizations */
-    priv = GET_PRIVATE(account);
-    if (!priv->parent)
-        return g_strdup("");
-
-    /* Figure out how much space is needed by counting the nodes up to
-     * the root. */
-    level = 0;
-    for (a = account; a; a = priv->parent)
-    {
-        priv = GET_PRIVATE(a);
-        level++;
-    }
-
-    /* Get all the pointers in the right order. The root node "entry"
-     * becomes the terminating nullptr pointer for the array of strings. */
-    names = (const gchar **)g_malloc(level * sizeof(gchar *));
-    names[--level] = nullptr;
-    for (a = account; level > 0; a = priv->parent)
-    {
-        priv = GET_PRIVATE(a);
-        names[--level] = priv->accountName;
-    }
-
-    /* Build the full name */
-    fullname = g_strjoinv(account_separator, (gchar **)names);
-    g_free(names);
-
-    return fullname;
+    auto path{gnc_account_get_all_parents (account)};
+    auto seps_size{path.empty() ? 0 : strlen (account_separator) * (path.size() - 1)};
+    auto alloc_size{std::accumulate (path.begin(), path.end(), seps_size,
+                                     [](auto sum, auto acc)
+                                     { return sum + strlen (xaccAccountGetName (acc)); })};
```

Comment:
You need more than one `seps_size` in the result:
```
    auto alloc_size{std::accumulate (path.begin(), path.end(), 0,
                                     [seps_size](auto sum, auto acc)
                                     { return sum + strlen (xaccAccountGetName (acc)) + seps_size; })};
```
The trailing `seps_size` conveniently makes at least enough space for the trailing `'\0'` and a few extra bytes does no harm.

Reply:
My original was correct; checked via `strlen()` and `alloc_size`. If we're not using GString reallocation then lets' fill pointers directly...

Reply:
Sorry, I misinterpreted `seps_size`.